extensions/parquet/table/src/main/java/io/deephaven/parquet/table/metadata/RowGroupInfoX.java

```diff
@@ -0,0 +1,273 @@
+
+    /**
+     * Splits evenly across {@code numRowGroups} RowGroups
+     */
+    private static class SplitEvenly extends RowGroupInfoX {
+        private final int numRowGroups;
+
+        private SplitEvenly(int numRowGroups) {
+            this.numRowGroups = numRowGroups;
+        }
+
+        @Override
+        public Table[] splitForRowGroups(@NotNull final Table input) {
+            return splitEvenly(input, numRowGroups);
+        }
+    }
+
+    /**
+     * Splits into RowGroups based on unique values within {@code groups}, and optionally further splits (per unique
+     * {@code groups} ensuring that no group is larger than {@code maxRows}. If {@code maxRows} is not desired, this
+     * parameter may be set to {@code Long.MAX_VALUE}
+     */
+    private static class SplitByGroups extends RowGroupInfoX {
+        final String[] groups;
+        final long maxRows;
+        final RowGroupInfoX maxRowsInfo;
+
+        private SplitByGroups(final long maxRows, final String[] groups) {
+            if (groups == null || groups.length == 0) {
+                throw new IllegalArgumentException("Requires at least one group");
+            }
+            this.groups = groups;
+            this.maxRows = maxRows;
+            this.maxRowsInfo = new SplitByMaxRows(maxRows);
+        }
+
+        private Table[] splitByGroups(@NotNull final Table input) {
+            final Table[] partitionedTables = input.partitionBy(groups).constituents();
+
+            // no need to split further if `maxRows` is greater than our original Table (which includes Long.MAX_VALUE)
+            if (maxRows >= input.size()) {
+                return partitionedTables;
+            }
+
+            int rowGroupCnt = 0;
+            final Table[][] maxSplit = new Table[partitionedTables.length][];
+            for (int ii = 0; ii < partitionedTables.length; ii++) {
+                maxSplit[ii] = maxRowsInfo.splitForRowGroups(partitionedTables[ii]);
+                rowGroupCnt += maxSplit[ii].length;
+            }
+
+            if (rowGroupCnt == partitionedTables.length) {
+                // all RowGroups already fit within `maxRows`
+                return partitionedTables;
+            }
+
+            // else we need to reconstitute the `Table[][]` into a straight `Table[]` of proper length
+            final Table[] retTables = new Table[rowGroupCnt];
+            int nextTargetIdx = 0;
+            for (final Table[] tables : maxSplit) {
+                System.arraycopy(tables, 0, retTables, nextTargetIdx, tables.length);
+                nextTargetIdx += tables.length;
+            }
+
+            return retTables;
+        }
+
+        @Override
+        public Table[] splitForRowGroups(@NotNull final Table input) {
+            final Table[] splitTables;
+            try (final SafeCloseable ignored = LivenessScopeStack.open()) {
+                splitTables = splitByGroups(input);
+                ensureConsistentOrdering(input, splitTables);
```

Comment:
This is not something we can use, we don't know that the keyspace of the resulting tables is the same as the input table.